### PR TITLE
Preview templates

### PR DESCRIPTION
**Pull request by GHaberis**
- Preview commercial templates added to assist developers when building templates
- **npm run serve-preview** will run preview on http://localhost:9000
- if making changes to preview **npm run build-preview** will rebuild preview resources
- dropdown populated by templates available in 'build' folder
- To preview a template you'll need to add a 'test.json' file under the relevant src directory, the keys of the JSON should match the placeholders in the HTML markup. The preview swaps the placeholders for the values on the JSON.

### CODE REVIEW

**Comment by GHaberis** [comment]
@regiskuckaertz @lps88 here's the preview for templates I built, this means you don't have to publish a template via DFP to see how it looks, it should hopefully speed development up.

File: preview/src/js/components/SelectTemplate.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    },
+
+    render: function () {
+        const listItems = this.props.templates.map(function (template) {
+            return (
+                <option value={template}>
+                    {template}
+                </option>
+            );
+        }.bind(this));
```
Comment (regiskuckaertz) [inline preview/src/js/components/SelectTemplate.js]: Iterator functions accept a second context argument.

**Comment by regiskuckaertz** [comment]
Lovely 👍

**Comment by annebyrne** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;xITXl42P&#x2F;24-epic-create-a-preview-app"> EPIC: Create a preview app</a>

File: preview/src/js/components/SelectTemplate.js (hunk 1 above)
Comment (GHaberis) [inline preview/src/js/components/SelectTemplate.js]: good to know, I removed the 'bind' anyway as the 'this' context wasn't even used in the function passed to map.

**Pull request merged**